flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperatorTest.java

```diff
@@ -783,6 +812,28 @@ public void testTimeoutCleanup() throws Exception {
         assertEquals(0, harness.getProcessingTimeService().getNumActiveTimers());
     }
 
+    @Test
+    public void testTimeoutAfterComplete() throws Exception {
+        StreamTaskMailboxTestHarnessBuilder<Integer> builder =
+                new StreamTaskMailboxTestHarnessBuilder<>(
+                                OneInputStreamTask::new, BasicTypeInfo.INT_TYPE_INFO)
+                        .addInput(BasicTypeInfo.INT_TYPE_INFO);
+        TimeoutAfterCompletionTestFunction.TIMED_OUT.set(false);
+        try (StreamTaskMailboxTestHarness<Integer> harness =
+                builder.setupOutputForSingletonOperatorChain(
+                                new AsyncWaitOperatorFactory<>(
+                                        new TimeoutAfterCompletionTestFunction(),
+                                        TIMEOUT,
+                                        1,
+                                        AsyncDataStream.OutputMode.UNORDERED))
+                        .build()) {
+            harness.processElement(new StreamRecord<>(1));
+            Thread.sleep(2 * TIMEOUT);
```

Comment:
I'm also not sure if this will prove to be stable - I don't have a concrete scenario, but generally speaking timeouts tend to brake.

Maybe replace it with:

```
while (timeOutHasNotHappened()) { 
  harness.processAll();
  Thread.sleep(1);
}
assertFalse(...);
asyncFunctionWaitLatch.complete();
```
?

Reply:
The test is testing that **no** timeout happens, so it's tricky to test.

I now had the idea to just register a timer in the test that is known to be processed after the timer under test. So we just wait until this timer has been processed and we should get a stable test. PTAL